### PR TITLE
[WIP] Refactor: Export function as arrow function const

### PR DESCRIPTION
**Pull request by AliMD**
This PR is WIP and needs your review to continue.

I am proud to get an opportunity to improve typescript `std` modules.
I have more than 16 years of experience in EcmaScript and will be happy if you allow me to make some improvement.

For example I think exporting functions as arrow syntax is better in ecmascript because of cost of function isolation like separate `this`, etc.

As seen in many popular projects like google polymer project
https://github.com/Polymer/lit-element/blob/master/src/lib/css-tag.ts#L87

Will you allow me to continue?
Thank you.

### CODE REVIEW

**Comment by AliMD** [comment]
@ry Could you please review this?
Your opinion is very important to me.
Thank you.

File: std/async/delay.ts [hunk 1]
```diff
@@ -1,9 +1,7 @@
 // Copyright 2018-2020 the Deno authors. All rights reserved. MIT license.
 /* Resolves after the given number of milliseconds. */
-export function delay(ms: number): Promise<void> {
-  return new Promise((res): number =>
-    setTimeout((): void => {
-      res();
-    }, ms)
-  );
+export const delay = (ms: number): Promise<void> => {
```
Comment (nayeemrmn) [inline std/async/delay.ts:3]: Again, the style guide clearly disallows this: https://deno.land/manual/contributing/style_guide#top-level-functions-should-not-use-arrow-syntax. If you want to know why, you're welcome to ask in the discord: https://discord.gg/deno _before_ opening lots of PRs.

**Comment by AliMD** [comment]
@nayeemrmn I'm sorry for disturbing
Please be a little kinder to me.
I'm really think use arrow syntax for export these types of functions ir much better
I would be grateful if we could talk about it.
Or you can just close this PR and I dont annoying you again with seperate PR.

**Comment by AliMD** [comment]
@kitsonk 
Thank you for your time
I was not convicted from technical view regarding TDZ, etc. And I think that I need to discuss this with the main contributors as you mentioned. I can guarantee that I can prevent TDZ.
If you think it is better to continue this in discord please close this PR.

**Comment by bartlomieju** [comment]
@AliMD thank you for the PR, but this change is in contradiction to our style guide. Closing without merge.

**Pull request closed without merging**